quack_app/lib/screens/friends/subscreens/add_friends_screen.dart

```diff
@@ -0,0 +1,97 @@
+
+class AddFriendsScreen extends StatefulWidget {
+  const AddFriendsScreen({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => AddFriendsScreenState();
+}
+
+class AddFriendsScreenState extends State<AddFriendsScreen> {
+  final _searchHistory = {
+    'bruce.lee@emory.edu': 'Bruce Lee',
+    'captain.marvel@emory.edu': 'Captain Marvel',
+    'wanda.vision@emory.edu': 'Wanda Vision'
+  };
+
+  String selectedTerm = "";
+  String currentEmail = "";
+  String currentName = "";
+  bool status = true;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        body: FloatingSearchBar(
+          hint: "Search for friends with email...",
+          openAxisAlignment: 0.0,
+          scrollPadding: const EdgeInsets.only(top: 16, bottom: 20),
+          elevation: 4.0,
+          onQueryChanged: (query) {
+            selectedTerm = query;
+            searchFriendResult();
+          },
+          onSubmitted: (query) {
+            selectedTerm = query;
+            searchFriendResult();
+          },
+          transitionCurve: Curves.easeInOut,
+          transitionDuration: const Duration(milliseconds: 500),
+          transition: CircularFloatingSearchBarTransition(),
+          debounceDelay: const Duration(milliseconds: 500),
+          actions: [
+            FloatingSearchBarAction(
+                showIfOpened: true,
+                child: CircularButton(
+                  icon: const Icon(Icons.search),
+                  onPressed: () {
+                    searchFriendResult();
+                    setState(() {});
+                  },
+                )),
+            FloatingSearchBarAction.searchToClear(
+              showIfClosed: false,
+            ),
+          ],
+          builder: (BuildContext context, Animation<double> transition) {
+            return ClipRRect(
+                child: Material(
+                    color: Colors.white,
+                    child: Container(
+                      height: 200.0,
+                      color: Colors.white,
+                      child: Column(
+                        children: [
+                          ListTile(
+                            trailing: Icon(status
+                                ? Icons.person_add
+                                : Icons.person_remove),
+                            title: Text(currentName),
+                            subtitle: Text(currentEmail),
+                            onTap: () {
+                              setState(() {
+                                status = !status;
+                              });
+                            },
+                          ),
+                        ],
+                      ),
+                    )
+                  )
+                );
+          },
+        ),
+        appBar: AppBar(title: const Text("Add Friends")));
+  }
+
+  void searchFriendResult() {
```

Comment:
Does this function call every time a character is changed in the search bar?

Reply:
I am honestly not sure. The whole onQueryChanged and onSubmit sections kinda threw me off a bit because they didn't really do what I thought they were going to do so I just kept that function in all of them in case there was a preference on where we wanted to have the search friends part

Reply:
Just looked at the documentation and onQueryChange is 'a callback that gets invoked when the input of the query inside the TextField changed' so it looks like this will get called each time the text is changed in anyway